kie-server-parent/kie-server-controller/kie-server-controller-websocket/src/main/java/org/kie/server/controller/websocket/management/WebSocketKieServerMgmtControllerImpl.java

```diff
@@ -51,6 +51,11 @@
     @PostConstruct
     public void configure() {
         LOGGER.info("Kie Server Controller Management WebSocket service initialized");
+        if (templateStorage == null && notificationService == null) {
+            // @PostConstruct can be called in Tomcat
+            LOGGER.info("templateStorage and notificationService are not injected but work with default implementations or loaded by ServiceLoader");
+            return;
+        }
```

Comment:
I suppose kie-server-controller in Tomcat doesn't rely on CDI. However `@PostConstruct` could be called. I haven't figured out why only the combination of Tomcat 9.0.50 + Java 8 causes the issue but applied the minimum (and safe) fix considering the release timeline.